MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -640,75 +649,84 @@ - (void)createAnalyzerFile {
 - (void)setupLogBuffer {
   @synchronized(self) {
 
-    // Array of 20 buffer file paths.
-    NSArray<NSString *> *bufferFiles = [self createLogBufferFilesIfNeeded];
+    // Array of 20 buffer file paths per priority.
+    // Each priority has 0.mscrasheslogbuffer ... 19.mscrasheslogbuffer.
+    for (NSInteger priority = 0; priority < kMSPriorityCount; priority++) {
+      NSArray *files = [self createBufferFilesIfNeededForPriority:(MSPriority)priority];
 
-    // Just to make sure we have max of 20 files on disk. If we have less than 20 files on disk, we can't really do
-    // anything. This case should never happen.
-    NSInteger count = bufferFiles.count >= ms_crashes_log_buffer_size ? ms_crashes_log_buffer_size : bufferFiles.count;
+      // Init the counter for each priority.
+      [self.bufferIndex setObject:@0 forKey:@(priority)];
 
-    // Initialize the bufferIndex and write pre-fill the filepaths of the MSCrashesBufferedLog
-    self.bufferIndex = 0;
-    for (NSUInteger i = 0; i < count; i++) {
-      MSCrashesBufferedLog emptyLog = MSCrashesBufferedLog{bufferFiles[i], nil};
-      msCrashesLogBuffer[i] = emptyLog;
+      // Create a buffer for the priority. Making use of `{}` as we're using C++11.
+      msCrashesLogBuffer[(MSPriority)priority][{}];
+      for (int i = 0; i < ms_crashes_log_buffer_size; i++) {
+        msCrashesLogBuffer[(MSPriority)priority][i] = MSCrashesBufferedLog{files[i], nil};
+      }
     }
   }
 }
 
-- (NSArray<NSString *> *)createLogBufferFilesIfNeeded {
-  NSArray *files = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:self.logBufferDir error:NULL];
-  NSMutableArray *logBufferFiles = [NSMutableArray arrayWithCapacity:ms_crashes_log_buffer_size];
+- (NSArray<NSString *> *)createBufferFilesIfNeededForPriority:(MSPriority)priority {
+  NSMutableArray<NSString *> *files = [NSMutableArray arrayWithCapacity:ms_crashes_log_buffer_size];
 
-  // Get already existing buffer files.
-  for (NSString *tmp in files) {
-    if ([[tmp pathExtension] isEqualToString:kMSLogBufferFileExtension]) {
-      NSString *filePath = [self.logBufferDir stringByAppendingPathComponent:tmp];
-      [logBufferFiles addObject:filePath];
-    }
-  }
+  // Create missing buffer files if needed. We don't care about which one's are already there,
+  // we'll just skip existing ones.
+  for (int i = 0; i < ms_crashes_log_buffer_size; i++) {
 
-  // Create missing buffer files if needed.
-  if (logBufferFiles.count < ms_crashes_log_buffer_size) {
-    NSInteger missingFileCount = ms_crashes_log_buffer_size - logBufferFiles.count;
-    for (int i = 0; i < missingFileCount; i++) {
-      NSString *logId = MS_UUID_STRING;
-      NSString *path = [self createBufferFileWithName:logId];
-      [logBufferFiles addObject:path];
-    }
+    // Files are named N.mscrasheslogbuffer where N is between 0 and ms_crashes_log_buffer_size.
+    NSString *logId = [NSString stringWithFormat:@"%ld", i];
```

Comment:
`NSString *logId = @(i).stringValue;` <-- faster and type-safer than invoking a full `stringWithFormat:`